### ISSUE TITLE
Dialog button focus and hover states are visually indistinct across themes

### ISSUE THREAD

**Issue by OptionalM** [zed-industries/zed#56533]
### Reproduction steps

1. Open a dialogue box with multiple buttons, for example by invoking "zed: import vs code settings"
2. Navigate the buttons via keyboard (up, down, tab)
3. Hover over the buttons with the mouse
4. Repeat with different (community) themes
5. Be confused about what exactly is happening

### Current vs. Expected behavior

**Current behavior**
<details>
<summary>Here's some examples. Can you tell which option is currently focused? What will happen if you press enter?</summary>
<b>Ayu Dark</b><br>
<img width="437" height="255" alt="Image" src="https://github.com/user-attachments/assets/e020e59b-3ce4-4365-8493-e26e21556a6e" /><br>
Mouse (not pictured) hovering "Cancel"<br>
<img width="427" height="242" alt="Image" src="https://github.com/user-attachments/assets/07f2e7cc-87d3-44ad-ac8c-62475e8f200f" /><br>
Mouse (not pictured) hovering "Ok"<br>
<img width="440" height="260" alt="Image" src="https://github.com/user-attachments/assets/7203e837-ad16-4c26-b0e4-b85cd81b79af" /><br>
<b>Gruvbox Dark</b><br>
<img width="417" height="102" alt="Image" src="https://github.com/user-attachments/assets/e5368a61-8ca7-4e35-8d67-5b7de3634440" /><br>
<b>One Dark</b><br>
<img width="421" height="105" alt="Image" src="https://github.com/user-attachments/assets/f23793bc-a144-4af5-acf6-79d52513481f" /><br>
<b>Catpuccin Frappé</b> (Community Extension 870k installs)<br>
<img width="420" height="102" alt="Image" src="https://github.com/user-attachments/assets/9cdfb127-fece-4ae0-aa79-8e5c9d7fad4f" /><br>
<b>Tokyo Night</b> (Community Extension 330k installs)<br>
<img width="421" height="101" alt="Image" src="https://github.com/user-attachments/assets/44d52d29-5be6-467e-a1eb-97a1fbbda81b" /><br>
<b>Zedokai</b> (Community Extension 90k installs)<br>
<img width="440" height="113" alt="Image" src="https://github.com/user-attachments/assets/193b37c8-33ab-412e-98c3-239d8102e76c" /><br>
<b>macOS Classic</b> (Community Extension 410k installs)<br>
<img width="431" height="102" alt="Image" src="https://github.com/user-attachments/assets/0b83271c-ddd8-488c-a98b-cf4ce6073a7c" />
</details>

You can also open the dialogue yourself and then (while leaving the dialogue open) open the theme switcher to easily see how different themes style these buttons. Sometimes it seems as if the selected button switches because the theming is so inconsistent.

**Expected behavior**
I'm not saying you should fix the community themes. But if there is this much inconsistency and so many themes are unclear, there is an underlying problem that needs fixing.

I'm no designer, but this seems common sense to me:
1. There should not be such an extreme visual difference between a mouse-hovered keyboard-selected button and a mouse-hovered not-keyboard-selected button (refer pictures 2 and 3)
2. The visual indicator for a keyboard-selected button should be more clear than the indicator for a mouse-hovered button (because you already see where the mouse is pointing).
3. Typically, a keyboard-focused element is highlighted with a border. If the element already had a border before then upon receiving focus its border should be thicker and/or have a more distinct color. In Zed it is the **exact opposite** across many (all?) themes: The border of the focused element is removed.

You can see all the expected behavior right here on Github: press tab to select inputs and buttons and see their borders turn bright blue. Hover these elements and notice how only their background (subtly) changes. Also notice how the mouse-hover state does not depend on the keyboard-focused state of the element.

### Zed version and system specs

Zed: v1.1.7+stable.268.c17f25cd782edffab64b7c4167abfa44f865b125 (Zed) 
OS: Linux Wayland fedora 44

**Comment by OptionalM** [comment]
Not a duplicate. This is about styling while that one is about functionality.

**Comment by yeskunall** [comment]
> I'm no designer, but this seems common sense to me:

Does it, though? 😅

> You can see all the expected behavior right here on Github

GitHub is a web application, owned by MSFT, which is HQ’d in the US. [The guidelines](https://www.federalregister.gov/documents/2024/04/24/2024-07758/nondiscrimination-on-the-basis-of-disability-accessibility-of-web-information-and-services-of-state) released by the Department of Justice under Title II of the Americans with Disabilities Act (ADA) have set a new bar for digital accessibility, which they have to embrace. Effective April 24, 2024, these standards require all web content to achieve compliance by April 24, 2026.

In other words, they kinda _have_ to comply with the WCAG standards.

Zed is not a web application and last I checked, we make no claims about meeting the Web Content Accessibility Guidelines (WCAG) 2.1, Level AA standards. That being said, we’re aware that we can do better with the contrast between our colors and overall the general accessibility of Zed. This has been discussed in the past, and very recently post 1.0 release. So it’s definitely on our radar.

**Comment by OptionalM** [comment]
(You can mark this comment as off-topic, because we basically agree.)

> >this seems common sense to me:
>
>Does it, though? 😅

Gnome, Thunderbird, Calibre, VLC, and KeePassXC are the non-electron (i.e. non-web) tools I have on hand right now and they all follow all three of these points. If not common sense then maybe a common pattern.

> GitHub [..] have to comply with [WCAG]

Mate, the issue template asks for "expected behavior" and I wanted to provide a visual example. Instead of drawing a picture I referenced GitHub for one way this can be done. You brought WCAG into this.

> we make no claims about meeting [..] AA

I just want to know what button I'm selecting. If you want to solve this in a WCAG-non-compliant way by adding an arrow with a rainbow animation to the focused button, by making this a dropdown or by not letting me cycle selections? I don't care. I think WCAG is a standard for a reason and if in doubt, maybe go that direction? But you also have a designer on staff, so if they come up with something else listen to them.

---

Sorry if I come off as stand-off-ish, but I spent more than half an hour drafting the issue and collecting screenshots to make it as clear and helpful as possible and it kind of hurt to have it so dismissed and misunderstood; your reply basically reads to me like a "Yeah you're kinda right, but also this is out of scope". If you care to explain, what made you think that I was asking for WCAG compliance or for you to follow the GitHub implementation specifically?

**Comment by OptionalM** [comment]
Related: #29399

**Comment by yeskunall** [comment]
> Yeah you're kinda right, but also this is out of scope

Nope -- you’re _absolutely_ right, and no, this is not out of scope. “This has been discussed in the past, and very recently post 1.0 release. So it’s definitely on our radar.” Even the thread you linked has a comment from me which basically says the same thing. No one is dismissing what you’re saying -- if that was the case this issue would be closed as not planned without a tailored response from me (or someone else). One could do that, too.

> GitHub for one way this can be done. You brought WCAG into this.

But you are missing the point. You used a web application as an example _first_. Everything that followed (my response to it, that is) is simply a way to let you and others know it is not right to treat a desktop application to the same standards as a web application. We support three platforms as of today, and thus have to account for the HCI/UI guidelines set forth by those platforms (if at they have [any](https://developer.apple.com/design/human-interface-guidelines/)). There will be hits and misses, and according to you (and others on various threads), we’re skewing towards more misses (right now). Again -- this is on our radar. Someone on the team may think this is something they can _and_ want to work on ATM, and so it’ll be picked up.

> Gnome, Thunderbird, Calibre, VLC, and KeePassXC are the non-electron (i.e. non-web) tools I have on hand right now and they all follow all three of these points.

Now you reference desktop applications, and also reference non-Electron based apps. This is what you should have referenced initially, and I would not have mentioned anything about WCAG guidelines.

FWIW -- WCAG _started_ out as specs for meeting certain criteria for those with disabilities using the Web, it does not apply to the Web alone these days.
